lib/mongodb-filter-wrapper.js

```diff
@@ -0,0 +1,157 @@
+		let modelFilters = {};
+
+		// Get model filters
+		if(model.constructor && model.constructor.fields)
+			modelFilters = model.constructor.fields;
+
+		// Validate filters given
+		if(typeof filters !== 'object' || (Array.isArray(filters) && filters.length === 0) || Object.entries(filters).length === 0)
+			return {};
+
+		return this.getParsedFilters(filters, modelFilters);
+	}
+
+	/**
+		* Validates filters given and loop calling to parse
+		* @returns parsedFilters Array|Object The filter parsed to use in MongoDB
+		* @param filters Object Filters defined as parameters
+		* @param modelFields Object Model that will use the filter
+		*/
+	static getParsedFilters(filters, modelFields) {
+		const orCondition = Array.isArray(filters);
+
+		let parsedFilters = orCondition ? [] : {};
+
+		// Only arrays as a filter
+		if(filters !== undefined && typeof filters === 'object' && Object.keys(filters).length > 0 && !Array.isArray(filters))
+			filters = [filters];
+
+		if(filters && filters.length > 0) {
+			// First loop over all filters given by parameters
+			filters.forEach(filterGiven => {
+				const fullyParsedFilter = this.parseObjectFilter(filterGiven, modelFields);
+				parsedFilters = (orCondition && Array.isArray(parsedFilters))
+					? [...parsedFilters, fullyParsedFilter] : Object.assign(parsedFilters, fullyParsedFilter);
+			});
+		}
+
+		if(orCondition && parsedFilters.length > 0) {
+			return {
+				$or: parsedFilters
+			};
+		}
+
+		return parsedFilters;
+	}
+
+	/**
+		* Parse the filter comparing given fields name and model fields
+		* @param filters Object Filters defined as parameters
+		* @param modelFields Object Model that will use the filter
+		* @return filterToResponse Object The filter parsed with all the values by one
+		*/
+	static parseObjectFilter(filters, modelFields) {
+		const filterToResponse = {};
+
+		for(const [field, value] of Object.entries(filters)) {
+			const fieldName = this.getFieldToCompare(field, modelFields);
+
+			const filter = (typeof filters[field] !== 'object') ? { value } : value;
+
+			filterToResponse[fieldName] = this.createFilter(filter, modelFields[field]);
+		}
+		return filterToResponse;
+	}
+
+	/**
+		* Creates the object or the value(string/array) to use in the filter
+		* @returns {string|Object} Value to use in filter, with type or if is an equal only the value to
+		* search
+		* @param filter Object Filters defined as parameters
+		* @param modelFields Object Model that will use the filter
+		*/
+	static createFilter(filter, modelFields) {
+		const type = this.getFilterType(filter, modelFields);
+		const value = this.getValue(filter);
+
+		if(type === 'equal')
+			return value;
+
+		return {
+			[this.filtersAllowed[type]]: value
+		};
+	}
+
+	/**
+		* Gets the value to use in the filter, if is a object(suppose ID) transform to string
+		* else search that exists the key value to return, or only returns an empty string
+		* @returns value String With the value to filter
```

Comment:
La tag `@returns` debe ir despues de los `@param` y debe especificar el _type_ (tipo de dato) en este caso sería:
`* @returns {string|number|object|array} The filter value`